pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -82,11 +82,20 @@ func SetDefaults_SeedNetworks(obj *SeedNetworks) {
 
 // SetDefaults_SeedSettingDependencyWatchdog sets defaults for SeedSettingDependencyWatchdog objects.
 func SetDefaults_SeedSettingDependencyWatchdog(obj *SeedSettingDependencyWatchdog) {
-	if obj.Endpoint == nil {
-		obj.Endpoint = &SeedSettingDependencyWatchdogEndpoint{Enabled: true}
+	// keeping the field `weeder` and `endpoint` in sync
```

Comment:
I don't think, the defaulting code is the right place. Instead, this should probably be done here: https://github.com/gardener/gardener/blob/c7ec5ac0ec9dbafe4fbfddc20e6758e3925dc141/pkg/registry/core/seed/strategy.go#L49-L69